pyMOE/gdsconverter.py

```diff
@@ -0,0 +1,282 @@
+                list_polygonsets.append(merged)
+                restart_merge = True
+    list_polygonsets.append(merged)
+    progress_bar(1)
+    
+    return list_polygonsets
+
+
+
+class GDSMask():
+    """
+    Class GDSMask:
+        Creates a class to integrate the GDS library and layout corresponding to a mask aperture.
+        Receives an aperture and provides methods to calculate the corresponding GDS layout
+    
+    Args:
+        mask:
+    
+    Methods:
+        mask
+        levels: number of unique discretized levels in mask 
+        layers: list of layers in layout
+        cells: list of cells in layout
+        total_polygons: total number of polygons in layout in all layers
+        total_vertices:  total number of vertices in all polygons in all layers
+
+        create_layout(): Creates layout (raster or vector or edges etc todo)
+        merge(layer): merges polygons in layer
+        plot(): plots mask
+        viewer(): Opens LayoutViewer of gdspy
+        save_gds(filename): saves layout to gds file
+
+    """
+    def __init__(self, mask, units=1e-6, precision=1e-9, verbose=True):
+        assert type(mask) is Aperture, "aperture must be of type Aperture"
+        self.mask = mask
+        self.gdslib = None
+        self.units = units
+        self.precision = precision
+        self.verbose=verbose
+        self._gdslib_init_error = "Error: gdsmask not created yet. Run GDSMask.create_layout()"
+        self.layers = None
+    
+    @property
+    def levels(self):
+        levels = self.mask.levels
+        assert levels is not None, "Cannot access GDSMask.levels as aperture is not yet discretized"
+        return self.mask.levels
+    
+    @property
+    def aperture(self):
+        aperturevar = self.mask.aperture_discretized
+        assert aperturevar is not None, "Cannot access GDSMask.aperture as aperture is not yet discretized"
+        return self.mask.aperture_discretized
+        
+    
+    @property
+    def total_polygons(self):
+        assert self.gdslib is not None, "%s"%self._gdslib_init_error
+        
+    @property
+    def total_vertices(self):
+        assert self.gdslib is not None, "%s"%self._gdslib_init_error
+    
+
+    
+    def _init_layout(self):
+        """ Initializes the gdslib"""
+        
+        self.gdslib = gdspy.GdsLibrary()
+        
+        
+    def view_layout(self):
+        """ Runs gdspy.LayoutViewer"""
+        
+        assert self.gdslib is not None, "Must create_layout() from discretized aperture first"
+        gdspy.LayoutViewer(self.gdslib)
+    
+    def write_gds(self, filename, cells=None, timestamp=None, binary_cells=None):
+        """ Writes layout to gds file using gdspy library"""
+        self.gdslib.write_gds(filename, cells, timestamp, binary_cells)
+        print("Saved %s"%(filename))
+        
+        
+        
+    def create_layout(self, mode="raster", cellname='mask', merge=True, break_vertices=250):
```

Comment:
From small tests merge = True takes more memory and time. 
Suggest to set merge = False by default (also, see comment on use of writer to file function while creating the raster file and/or use of instatiation. Merge function can be used a posteriori on the generated files).

Reply:
Agreed. changed to default=False